### PR TITLE
v1.14 Backports 2023-12-13

### PR DESCRIPTION
**Pull request by giorio94**
* [x] #27894 (@leblowl)
    - :information_source: Introduced a simplified version of the `identity.IsWorld()` method, which was not present in v1.14, as added in a94fa56f6713 ("Fix CIDR to World Entity Conversion Bug")
 * [x] #29455 (@mhofstetter)
   - :information_source: Skipped changes in tests-e2e-upgrade.yaml, as not present in v1.14.
 * [ ] ~~#29625 (@tklauser)~~ 
   - :x: Dropped based on https://github.com/cilium/cilium/pull/29863#issuecomment-1854108601.
 * [x] #29694 (@mhofstetter)
 * [x] #29201 (@tommyp1ckles)
   - :information_source: Dropped the Registry parameter from the new invoke function in `pkg/metrics/cell.go`, as it is not provided.
 * [x] #29448 (@giorio94)
 * [x] #29675 (@giorio94)
 * [x] #29670 (@giorio94)
 * [x] #29752 (@lmb)
 * [x] #29502 (@mhofstetter)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 27894 29455 29694 29201 29448 29675 29670 29752 29502
```

### CODE REVIEW

**Comment by giorio94** [comment]
/test-backport-1.14

File: pkg/fqdn/name_manager.go
Comment (tklauser) [inline pkg/fqdn/name_manager.go]: We need to replace this by `golang.org/x/exp/slices` in `v1.14` for Go 1.20 which doesn't have the standard library `slices` package yet.

**Review by tklauser — CHANGES_REQUESTED** [review]

**Review by mhofstetter — COMMENTED** [review]
I realized that I forgot to label https://github.com/cilium/cilium/pull/29502 with `needs-backport/v1.14`. This fix is needed in addition to https://github.com/cilium/cilium/pull/29455 . Would be great if we could include this. Even though it's again kind of depending on the LVH images that are used in v1.14.

**Comment by giorio94** [comment]
> I realized that I forgot to label #29502 with `needs-backport/v1.14`. This fix is needed in addition to #29455 . Would be great if we could include this. Even though it's again kind of depending on the LVH images that are used in v1.14.

Added, PTAL

**Comment by tklauser** [comment]
> [fqdn: avoid converting from `netip.Addr` to `net.IP` and back #29625](https://github.com/cilium/cilium/pull/29625) (@tklauser)
> * ⚠️ MapSelectorsToIPsLocked is still returning a map value is a net.IP slice; hence, let's convert the netip.IPAddr  on the fly before assigning it. @tklauser please let me know if you prefer this approach or drop the backport of this PR.

Ah, I see `MapSelectorsToIPsLocked` only got changed in #29036 which isn't backported (and probably won't be). Without this change backporting my PR is kind of pointless. Let's drop it for now.

**Review by mhofstetter — APPROVED** [review]
My commits look good. Thanks @giorio94 

Let's :pray: that the tests succeed :)

**Comment by giorio94** [comment]
> Ah, I see `MapSelectorsToIPsLocked` only got changed in #29036 which isn't backported (and probably won't be). Without this change backporting my PR is kind of pointless. Let's drop it for now. Sorry for the trouble caused.

Dropped, thanks for confirming.

**Comment by giorio94** [comment]
/test-backport-1.14

**Review by tklauser — APPROVED** [review]
I tried removing myself from the reviewer list after my PR was [dropped](https://github.com/cilium/cilium/pull/29863#issuecomment-1854123306) but that seems to cause my [requested changes](https://github.com/cilium/cilium/pull/29863#pullrequestreview-1779846982) to block the PR again. Approving to unblock this PR.

**Review by lmb — APPROVED** [review]

**Comment by giorio94** [comment]
Both Conformance E2E and Conformance Cluster Mesh failures appear to be due to slow image retrieval from quay.
One of the Conformance Ginkgo matrix entries failed during LVH VM setup.

**Review by gandro — APPROVED** [review]
Thank you, approving https://github.com/cilium/cilium/pull/27894

**Review by viktor-kurchenko — APPROVED** [review]

**Review by tommyp1ckles — APPROVED** [review]

**Comment by giorio94** [comment]
> Both Conformance E2E and Conformance Cluster Mesh failures appear to be due to slow image retrieval from quay.

The same happened for one of the Conformance IPsec E2E matrix entries. Currently rerunning

**Comment by giorio94** [comment]
CI is green. Marking ready to merge as it only misses the final approval from @cilium/tophat, who will be merging the PR.

**Pull request merged**
